_data-prepper/common-use-cases/text-processing.md

```diff
@@ -0,0 +1,215 @@
+  "MONTH":"Oct",
+  "YEAR":"2000",
+  "response_status":200,
+  "HOUR":"13",
+  "TIME":"13:55:36",
+  "MINUTE":"55",
+  "SECOND":"36",
+  "IPORHOST":"198.126.12",
+  "MONTHDAY":"10",
+  "INT":"-0700",
+  "timestamp":"10/Oct/2000:13:55:36 -0700"
+}
+```
+{% include copy-curl.html %}
+
+Note that the `IPORHOST` capture now shows up as a new key, along with some internal unnamed captures like `MONTH` and `YEAR`. The `HTTPDATE` keyword is using these patterns, which you can see in the default patterns file.
+
+## Overwriting keys
+
+Include the `keys_to_overwrite` option to specify which existing keys of a record to overwrite if there's a capture with the same key value.
+
+For example, you can modify the preceding Grok configuration to replace `%{NUMBER:response_status:int}` with `%{NUMBER:message:int}` and add `message` to the list of keys to overwrite.
+
+```json
+processor:
+  - grok:
+      match:
+        keys_to_overwrite: ["message"]
+        message: ['%{IPORHOST:clientip} \[%{HTTPDATE:timestamp}\] %{NUMBER:message:int}']
+```
+{% include copy-curl.html %}
+
+In the resulting grokked log, the original message is overwritten with the number `200`.
+
+```json
+{ 
+  "message":200,
+  "clientip":"198.126.12",
+  "timestamp":"10/Oct/2000:13:55:36 -0700"
+}
+```
+{% include copy-curl.html %}
+
+## Using custom patterns
+
+Include the `pattern_definitions` option in your Grok configuration to specify custom patterns.
+
+The following configuration creates custom regex patterns named `CUSTOM_PATTERN-1` and `CUSTOM_PATTERN-2`. By default, the plugin continues until it finds a successful match.
+
+```json
+processor:
+  - grok:
+      pattern_definitions:
+        CUSTOM_PATTERN_1: 'this-is-regex-1'
+        CUSTOM_PATTERN_2: '%{CUSTOM_PATTERN_1} REGEX'
+      match:
+        message: ["%{CUSTOM_PATTERN_2:my_pattern_key}"]
+```
+{% include copy-curl.html %}
+
+If you specify `break_on_match` as `false`, the pipeline tries to match all patterns and extract keys from the incoming events:
+
+```json
+processor:
+  - grok:
+      pattern_definitions:
+        CUSTOM_PATTERN_1: 'this-is-regex-1'
+        CUSTOM_PATTERN_2: 'this-is-regex-2'
+        CUSTOM_PATTERN_3: 'this-is-regex-3'
+        CUSTOM_PATTERN_4: 'this-is-regex-4'
+      match:
+        message: [ "%{PATTERN1}”, "%{PATTERN2}" ]
+        log: [ "%{PATTERN3}", "%{PATTERN4}" ]
+        break_on_match: false
+```
+{% include copy-curl.html %}
+
+You can define your own custom patterns to use for pattern matching in pipelines. In the previous example, `my_pattern` will be extracted after matching the custom patterns.
+
+## Storing captures with a parent key
+
+Include the `target_key` option in your Grok configuration to wrap all captures for a record in an additional outer key value.
+
+For example, you can modify the preceding Grok configuration to add a target key named `grokked`.
+
+```json
+processor:
+   - grok:
+       target_key: "grok"
```

Comment:
@dlvenable Should this be "grokked", as mentioned in preceding sentence?

Reply:
@vagimeli , Yes, you are correct. I missed that typo in my first review.